File: 03b.py
#!/usr/bin/python3

import re

with open('03.dat') as f:
    data = f.read()
instruction_lines = data.split('\n')
instruction_lines.pop()
triangles = 0
counter = 0;
candidates = [[] for _ in range(3)]
for line in instruction_lines:
    counter += 1
    sides = re.sub(r'\s+', ' ', line).strip().split(' ')
    for key, value in enumerate(sides):
        candidates[key].append(value)

    if counter % 3 == 0:
        for candidate in candidates:
            is_triangle = True
            for side in candidate:
                other_sides = candidate.copy()
                other_sides.remove(side)
                side_sum = sum([int(x) for x in other_sides])
                if side_sum <= int(side):
                    is_triangle = False
                    break
            if is_triangle:
                triangles += 1
        for i, _ in enumerate(candidates):
            candidates[i] = []
print(triangles)

File: 03b.php
<?php

$file = "03.dat";
$fin = fopen($file, "r");
$data = fread($fin, filesize($file));
$instruction_lines = preg_split('/\n/', $data, -1, PREG_SPLIT_NO_EMPTY);
$triangles = 0;
$counter = 0;
$candidates = [[], [], []];
foreach ($instruction_lines as $line) {
    $counter++;
    $sides = explode(' ', trim(preg_replace('/\s+/', ' ', $line)));
    for ($i=0; $i<count($sides); $i++) {
        array_push($candidates[$i], $sides[$i]);
    }

    if ($counter % 3 == 0) {
        foreach ($candidates as $candidate) {
            $is_triangle = true;
            foreach ($candidate as $side) {
                $other_sides = $candidate;
                array_splice($other_sides, array_search($side, $other_sides), 1);
                $side_sum = array_reduce($other_sides, fn($sum, $other_side) => intval($sum) + intval($other_side));
                if ($side_sum <= intval($side)) {
                    $is_triangle = false;
                    break;
                }
            }
            if ($is_triangle) {
                $triangles += 1;
            }
        }

        for ($i=0; $i<count($candidates); $i++) {
            $candidates[$i] = [];
        }
    }
}
echo $triangles;

fclose($fin);

?>
